Azure/communication-ui-library-ios
# [Bugfix] Fix adding duplicate messages to message repository

## Purpose

* When sending out a message, we append the new message when message is still sending and then replace it's id when sent. However, this will also trigger addReceivedMessage which will append this message to repository as well.
<img src="https://user-images.githubusercontent.com/107075081/204967985-d8a5f9bc-b9a7-47ce-9ef8-ab74ae001275.png" width=700 height=200>

* This will cause message send status not displaying correctly because the new message added by `addReceivedMessage` doesn't have the right sendStatus

## Does this introduce a breaking change?



## Pull Request Type
What kind of change does this Pull Request introduce?



## Pull request checklist

This PR has considered:

## How to Test

* After line 108 in `MessageListViewModel`, add the following lines:
```
            if messageRepositoryManager.messages.count > 1 {
                print("messages count: \(messages.count), manager messages count: ",
                      messageRepositoryManager.messages.count)
                let lastManagerMessage =
                messageRepositoryManager.messages[messageRepositoryManager.messages.count - 1]
                print("last manager message: \(lastManagerMessage.id), content: ",
                      lastManagerMessage.content)
                let secondLastManagerMessage =
                messageRepositoryManager.messages[messageRepositoryManager.messages.count - 2]
                print("second last manager message: \(secondLastManagerMessage.id), content: ",
                      secondLastManagerMessage.content)
```
* Send out a message
* See the printed messages
